src/actions/notificationActions.js

```diff
@@ -0,0 +1,91 @@
+export const NOTIFICATIONS_READ = 'NOTIFICATIONS_READ'
+export const EMIT_NOTIFICATION = 'EMIT_NOTIFICATION'
+export const NOTIFICATION_EMITTED = 'NOTIFICATION_EMITTED'
+export const TOGGLE_RECEIVE_PUSH_NOTIFICATIONS = 'TOGGLE_RECEIVE_PUSH_NOTIFICATIONS'
+export const NOTIFICATION_CLICKED = 'NOTIFICATION_CLICKED'
+
+export const extensionMessages = {
+  popupOnLoad: 'SIA_EXTENSION_POPUP_ON_LOAD'
+}
+
+export const notificationClicked = (notificationId, buttonIndex) => ({
+  type: NOTIFICATION_CLICKED,
+  notificationId,
+  buttonIndex
+})
+
+export const notificationEmitted = (eventId, notificationId, buttonActions) => ({
+  type: NOTIFICATION_EMITTED,
+  eventId,
+  notificationId,
+  buttonActions
+})
+
+export const emitNotification = (notification) => ({
+  type: EMIT_NOTIFICATION,
+  notification
+})
+
+export const toggleReceivePushNotifications = (enable) => ({
+  type: TOGGLE_RECEIVE_PUSH_NOTIFICATIONS,
+  enable
+})
+
+export const notificationsRead = { type: NOTIFICATIONS_READ }
+
+export const isChromeExtension = (() => chrome &&
+                                        chrome.extension &&
+                                        chrome.extension.getBackgroundPage)()
+
+export const isChromeExtensionBackground = (() => isChromeExtension &&
+                                                  chrome.extension.getBackgroundPage() === window)()
+
+export const isChromeExtensionPopup = (() => isChromeExtension &&
+                                             chrome.extension.getBackgroundPage() !== window)()
+
+export const registerExtensionActions = (dispatch) => {
```

Comment:
Generally things that are 'do-once' aren't really 'actions'; it might be best to move this to a 'notificationsService'. Compare to signalRService which imports signalRActions and registers events that dispatch those actions.

Reply:
Will take a look.